src/routes/safe/components/Apps/components/AppFrame.tsx

```diff
@@ -1,92 +1,297 @@
+import { useSelector } from 'react-redux'
+import {
+  INTERFACE_MESSAGES,
+  Transaction,
+  RequestId,
+  LowercaseNetworks,
+  SendTransactionParams,
+} from '@gnosis.pm/safe-apps-sdk'
+
+import {
+  safeEthBalanceSelector,
+  safeParamAddressFromStateSelector,
+  safeNameSelector,
+} from 'src/logic/safe/store/selectors'
+import { grantedSelector } from 'src/routes/safe/container/selector'
+import { getNetworkName } from 'src/config'
 import { SAFELIST_ADDRESS } from 'src/routes/routes'
+import { isSameURL } from 'src/utils/url'
+import { useAnalytics, SAFE_NAVIGATION_EVENT } from 'src/utils/googleAnalytics'
+import { loadFromStorage, saveToStorage } from 'src/utils/storage'
+import { staticAppsList } from 'src/routes/safe/components/Apps/utils'
+
+import ConfirmTransactionModal from '../components/ConfirmTransactionModal'
+import { useIframeMessageHandler } from '../hooks/useIframeMessageHandler'
 import { useLegalConsent } from '../hooks/useLegalConsent'
-import { SafeApp } from '../types'
 import LegalDisclaimer from './LegalDisclaimer'
+import { APPS_STORAGE_KEY, getAppInfoFromUrl } from '../utils'
+import { SafeApp, StoredSafeApp } from '../types.d'
 
-const StyledIframe = styled.iframe`
-  padding: 15px;
-  box-sizing: border-box;
+const LoadingContainer = styled.div`
   width: 100%;
   height: 100%;
+  display: flex;
+  align-items: center;
+  justify-content: center;
 `
 
-const LoadingContainer = styled.div`
-  width: 100%;
-  height: 100%;
+const OwnerDisclaimer = styled.div`
   display: flex;
   align-items: center;
   justify-content: center;
+  flex-direction: column;
+  height: 476px;
+`
+
+const AppWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+`
+
+const StyledCard = styled(Card)`
+  flex-grow: 1;
 `
 
-const IframeWrapper = styled.div`
-  position: relative;
+const StyledIframe = styled.iframe`
   height: 100%;
   width: 100%;
-  overflow: hidden;
+  overflow: auto;
+  box-sizing: border-box;
 `
 
-const Centered = styled.div`
-  display: flex;
-  align-items: center;
-  justify-content: center;
-  flex-direction: column;
+const Breadcrumb = styled.div`
+  height: 51px;
 `
 
-type AppFrameProps = {
-  selectedApp: SafeApp | undefined
-  safeAddress: string
-  network: string
-  granted: boolean
-  appIsLoading: boolean
-  onIframeLoad: () => void
+type ConfirmTransactionModalState = {
+  isOpen: boolean
+  txs: Transaction[]
+  requestId?: RequestId
+  params?: SendTransactionParams
+}
+
+type Props = {
+  appUrl: string
+}
+
+const NETWORK_NAME = getNetworkName()
+
+const INITIAL_CONFIRM_TX_MODAL_STATE: ConfirmTransactionModalState = {
+  isOpen: false,
+  txs: [],
+  requestId: undefined,
+  params: undefined,
 }
 
-const AppFrame = forwardRef<HTMLIFrameElement, AppFrameProps>(function AppFrameComponent(
-  { selectedApp, safeAddress, network, appIsLoading, granted, onIframeLoad },
-  iframeRef,
-): React.ReactElement {
+const AppFrame = ({ appUrl }: Props): React.ReactElement => {
```

Comment:
AppFrame was meant to be a component that is responsible for displaying the app inside an iframe and not hold any logic. Why did you move the logic here?

Reply:
Before this change, separating the logic made sense, mixing the app list logic with the app frame was chaos. Now App frame is rendered in a new URL and the logic here is merely to determine what to render.

Reply:
>  mixing the app list logic with the app frame

app list logic was hidden inside useAppList hook, wasn't it?

Reply:
yes, but we don't want to make use of useAppList in AppFrame, it will load the complete app list for no reason. 
That's why I moved to AppFrame the `removeApp` functionality. 
I can create another hook for that functionality if you think, we can also move the logic that checks if an app.url exits in the static list